src/transforms/remap.rs

```diff
@@ -133,14 +135,50 @@ pub struct RemapConfig {
     #[configurable(derived, metadata(docs::hidden))]
     #[serde(default)]
     pub runtime: VrlRuntime,
+
+    #[configurable(derived, metadata(docs::hidden))]
+    #[serde(skip)]
+    #[derivative(Debug = "ignore")]
+    pub cache: Mutex<
+        Vec<(
+            (TableRegistry, schema::Definition),
+            std::result::Result<(Program, String, MeaningList), String>,
+        )>,
```

Comment:
It would be helpful to have some commentary here on why this `Vec`, which is used as a lookup table, can't be a `BTreeMap` or `HashMap`.